### DISCUSSION TITLE
cookie is not setting in Server side using App router

### DISCUSSION THREAD

**Discussion by Arunjenson-ss** [andreizanik/cookies-next#52] [General]
setCookie is not working in App router (next13)

**Comment by greg2012201** [comment]
I think I could work on it. But I'm waiting for accepting my previous PR with middleware support(https://github.com/andreizanik/cookies-next/pull/51).

**Comment by Arunjenson-ss** [comment]
@greg2012201 when can we except your PR to be merged?

**Reply by greg2012201** [reply]
I'm waiting for @andreizanik. Btw. I think that setting cookies in RSC could be possible using cookies imported from next/headers. If it works I will try implement it

**Comment by Arunjenson-ss** [comment]
does any other package provide with the server side cookie setting?

**Reply by greg2012201** [reply]
I haven't heard about any package with that kind of functionality.

**Comment by andreizanik** [comment]
@Arunjenson-ss, can you check version 4.0.0? 

@greg2012201 added support in middleware

**Reply by greg2012201** [reply]
I think that it still doesn't work. The middleware support should work with middleware and route handlers. I will work on next PR with RSC support. Stay tuned 😜
